Improvements and structure changes for autoscaling docs (#2439)
knative/docs: docs/serving/autoscaling/target-burst-capacity.md

---
title: "Target burst capacity"
linkTitle: "Target burst capacity"
weight: 50
type: "docs"
---

_Target burst capacity_ is a [global and per-revision](./global-vs-per-revision.md) integer setting.

This value specifies how large of a _burst_ of traffic a Knative application can handle without buffering.

> If a burst of traffic is too large for the application to handle, the Knative _Activator_ will be placed in the request path to protect the application revision and optimize request load balancing.

Reviewer: ```suggestion
If a desired burst of traffic is too large for the application to handle at current scale, the Knative _Activator_ will be placed in the request path to protect the application revision and optimize request load balancing.
```

Author: It's actually in the Google developer docs style guide that we shouldn't use "desire", and I think this would sort of just add fluff anyway.

> <!-- Question: is revision accurate here or is it protecting replicas of an app?-->

Reviewer: I think revision==app, kind of.

> <!-- TODO: docs about Activator?-->

Author: @vagababov

Reviewer: Activator for most intents and purposes is detail of implementation here. So I am not sure we want to add more docs here.

If a traffic burst is too large for the application to handle, the _Activator_ service will be placed in the request path to protect the revision and optimize request load balancing.

The Activator service is responsible for receiving and buffering requests for inactive revisions, or for revisions where a traffic burst is larger than the limits of what can be handled without buffering for that revision.

Target burst capacity can be configured using a combination of the following parameters:

* Setting the targeted concurrency limits for the revision. For more information, see the documentation on [concurrency](./concurrency.md).
* Setting the target utilization parameters. For more information, see the documentation on [target utilization](./concurrency.md#target-utilization).
* Setting the target burst capacity per revision.

## Setting the target burst capacity per revision

* **Global key:** No global key.
* **Per-revision annotation key:** `autoscaling.knative.dev/targetBurstCapacity`
* **Possible values:** float
* **Default:** `70`

> **Note:** If the activator is in the routing path, it will fully load all replicas up to `containerConcurrency`. It currently does not take target utilization into account.

Reviewer: ```suggestion
**Note:** If the activator is in the routing path, it will fully load all replicas up to `containerConcurrency`. It currently applies target utilization only on revision level.
```
Or somewhere thereof. Since the integral utilization is respected, per pod one isn't

* Setting the target burst capacity per revision.

## Setting the target burst capacity per revision

* **Global key:** No global key.
* **Per-revision annotation key:** `autoscaling.knative.dev/targetBurstCapacity`
* **Possible values:** float
* **Default:** `70`

**Note:** If the activator is in the routing path, it will fully load all replicas up to `containerConcurrency`. It currently does not take target utilization into account.

**Example:**
{{< tabs name="targetBurstCapacity" default="Per Revision" >}}
{{% tab name="Per Revision" %}}
```yaml
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  annotations:
  name: s3
  namespace: default
spec:
  template:
    metadata:
      annotations:
        autoscaling.knative.dev/minScale: "2"
        autoscaling.knative.dev/targetBurstCapacity: "70"
    spec:
      containerConcurrency: 4
      containers:
      - env:
        - name: TARGET
          value: T2
        image: "your docker image"
        imagePullPolicy: Always
        name: user-container
      timeoutSeconds: 300

> ```

Reviewer 2: Let's reuse the same example as for all other examples. This contains a lot of fields that are not required at all. Only the `targetBurstCapacity` setting is significant for these docs.
